registry.go

```diff
@@ -104,6 +105,11 @@ type Layer interface {
 
 	// CreatedAt returns the time this layer was created.
 	CreatedAt() time.Time
+
+	// ServeHTTP allows a layer to serve itself, whether by providing
+	// a redirect directly to the content, or by serving the content
+	// itself
+	ServeHTTP(w http.ResponseWriter, r *http.Request)
```

Comment:
Alternatively the blob can write an appropriate error about why it can't serve the request within its ServeHTTP. Given the minimal changes this required in the handlers.LayerHandler it feels very safe. It won't be a problem to change it later so maybe we can reassess if we manage to collapse the layer and manifest code into the single blob concept.